code/_globalvars/regexes.dm

```diff
@@ -0,0 +1,7 @@
+//These are a bunch of regex datums for use /((any|every|no|some|head|foot)where(wolf)?\sand\s)+(\.[\.\s]+\s?where\?)?/i
+GLOBAL_DATUM_INIT(is_http_protocol, /regex, regex("^https?://"))
+
+GLOBAL_DATUM_INIT(is_website, /regex, regex("http|www.|\[a-z0-9_-]+.(com|org|net|mil|edu)+", "i"))
+GLOBAL_DATUM_INIT(is_email, /regex, regex("\[a-z0-9_-]+@\[a-z0-9_-]+.\[a-z0-9_-]+", "i"))
+GLOBAL_DATUM_INIT(is_alphanumeric, /regex, regex("\[a-z0-9]+", "i"))
+GLOBAL_DATUM_INIT(is_punctuation, /regex, regex("\[.!?]+", "i"))
```

Comment:
could you make these procs instead that return the regexes as needed so we don't have theses around 100% of the time when they aren't use 99% of it

Reply:
Yeah, that will solve some possible issues. I wonder which has the lower memory footprint...